tests/integration/views/test_request.py

```diff
@@ -443,6 +443,28 @@ def test_request_submit_not_author(airlock_client):
     assert persisted_request.status == RequestStatus.PENDING
 
 
+def test_request_workspace(airlock_client):
+    airlock_client.login(workspaces=["test1"])
+    other_user = factories.create_user("other", ["test1"], False)
+
+    release_request1 = factories.create_release_request(
+        "test1", user=other_user, status=RequestStatus.PENDING
+    )
+    factories.write_request_file(release_request1, "group", "path/test.txt")
+
+    release_request2 = factories.create_release_request(
+        "test1", user=other_user, status=RequestStatus.PENDING
+    )
```

Comment:
I'm surprised that a workspace can have two pending requests for the same user.  Have I misunderstood something?

Reply:
that is a mistake - I assume that this is only possible because we've going via the factory route & this would not be possible in production. Is it worth adding a safety mechanism to `factories.create_release_request`?

Reply:
I'd strongly prefer it if it was impossible to put the system in an invalid state in tests.

A safety mechanism in the factory would be one solution to this specific problem, but better would be if test objects were created via the same route that production objects are created.  That way, any invariants that are enforced by the production code are also enforced in tests.  I've not looked into how we'd do that here but would be happy to pair.

Reply:
just looking at this - I think the production objects are created via `bll.get_or_create_current_request()` which means that you simply can't create a second request on a workspace (as a given user) because you would get the first one. The test objects are created by `_create_release_request` which obviously bypasses this, but also has some extra args that let us do useful things for writing tests - i.e. creating new requests in certain states (e.g. `RequestStatus.SUBMITTED`). 

One possible refactor would be:

* `_create_current_request` becomes `_get_or_create_current_request` which gets a request if a matching one exists - we would match using the same criteria as the current `get_or_create_current_request`, i.e. just the workspace-user pair, not e.g. `status`, I would guess this might cause an existing test or two to fail
* `get_or_create_current_request` then becomes a dumb method that calls `_get_or_create_current_request` with the correct defaults

What do you think?

Reply:
have amended this test to create the two release requests attached to two different users. we can look at the wider question of inviolates in testing elsewhere.